06_puppy/jimbotech/sync_mapstore.go

```diff
@@ -49,13 +49,13 @@ func (s *SyncMapStore) ReadPuppy(id uint32) (*Puppy, error) {
 // UpdatePuppy threadsafe update your puppy store.
 func (s *SyncMapStore) UpdatePuppy(id uint32, puppy *Puppy) error {
 	_, found := s.Load(id)
-	if found {
-		puppy.ID = id
-		sp := *puppy
-		s.Store(id, &sp)
-		return nil
+	if !found {
```

Comment:
minor nitpick: you could change this to `if _, found := s.Load(id); !found {`

And while `found` is fine as a name, the name `ok` is used far more often, so much so that it is called the "comma ok idiom" - `if _, ok := s.Load(id); !ok {` - for a while "found" seems like a better name because it is more descriptive, but after a while, "ok" becomes standard (it is also used when receiving from channels: `v, ok := <-ch`). So you'll see "ok" used a lot here.